servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancerReadySubscriber.java

```diff
@@ -38,13 +38,18 @@ final class LoadBalancerReadySubscriber extends DelayedCancellable implements Su
 
     /**
      * Get {@link Completable} that will complete when a {@link LoadBalancerReadyEvent} returns {@code true}
-     * from {@link LoadBalancerReadyEvent#isReady()}.
+     * from {@link LoadBalancerReadyEvent#isReady()} or {@code null} if the {@link LoadBalancer} is already ready.
+     * <p>
+     * This can happen if the {@link LoadBalancer} already signaled its {@link LoadBalancerReadyEvent ready event}, but
+     * all hosts become unhealthy later.
+     *
      * @return A {@link Completable} that will complete when a {@link LoadBalancerReadyEvent} returns {@code true}
-     * from {@link LoadBalancerReadyEvent#isReady()}.
+     * from {@link LoadBalancerReadyEvent#isReady()} or {@code null} if the {@link LoadBalancer} is already ready.
      */
-    public Completable onHostsAvailable() {
+    @Nullable
+    Completable onHostsAvailable() {
         Processor onHostsAvailable = this.onHostsAvailable;
-        return onHostsAvailable == null ? completed() : fromSource(onHostsAvailable);
+        return onHostsAvailable == null ? null : fromSource(onHostsAvailable);
```

Comment:
I'm not sure if `null` is a great signal here that something is ready - the old behavior (completing immediately) seemed like a clear signal that something is available.

For my understanding, it seems that there are two different signals "merged" here:

-  Load balance is ready (LoadBalancerReadyEvent#isReady())
-  Hosts are available

So if my understanding is correct the load balancer can become ready only once (so completable semantics would suffice) vs. hosts can become available, then unavailable, then available again. If that is true, would it make sense to turn hostsAvailable from a Completable into a Publisher which emits state? (Available / Unavailable) and caches the last one? So it doesn't matter if one subscribes to the hot publisher, they would just add a .takeUntil(e == available) or similar.

Reply:
`onHostsAvailable` is the same as LB publisher emits "ready" event. 
LB can turn into "not-ready" state at any time late if SD propagated all "unavailable" events.
We can not use `Publisher` here because we have to translate publisher events from LB to `Completable`, bcz this is what is used as a retry signal.

Reply:
Alternative proposal is in #2648

Reply:
@daschl Thanks for the question! It helped to find another way that does not require changes for existing `LoadBalancerReadySubscriber`